Upstream libraries to latest.

Kichura opened the pull request:
Libraries that will be updated in build.gradle:

Kohsuke's GitHub API: 1.114 -> 1.313,
Proguard-Gradle: 7.2.0-beta2 + 7.1.0 -> 7.3.0-beta2 ~~+ 7.2.0~~,
QuiltMC's Loom: 0.12.+ -> 1.0.+,
Johnrengelman's shadow: 7.0.0 -> 7.1.2,
Night-Config: 3.6.5 -> 3.6.6,
Junit-Bom: 5.7.2 -> 5.9.1.

~~Libraries that will be updated in gradle.properties:
ASM: 9.3 -> 9.4,~~

Libraries that will be updated in quilt_installer.json:

Tiny-Remapper: 0.8.1 -> 0.8.6,
Quilt-JSON5: 1.0.1 -> 1.0.2,
~~ASM: 9.3 -> 9.4.~~

Libraries that will be updated in gradle-wrapper.properties:

Gradle: 7.3.3 -> 7.5.1.

Libraries that will be updated in GitHub actions:

Ubuntu: 20.04 -> 22.04,
Checkout: v1 -> v3.

*(Edits are welcomed - The Loom and JSON5 contexts can be undoed if required; aside from that, should normally run fine with these changes unless it breaks in older versions of MC itself)*

Kichura committed:
Upstream libraries to latest.

triphora commented on build.gradle:
```diff
@@ -85,17 +85,17 @@ dependencies {
 	/*include*/ api 'org.quiltmc:quilt-json5:1.0.1'
```
This needs to be changed too

Kichura replied:
Fixed.

OroArmor commented:
I would rather a system be implemented to automatically edit the `installer.json` file rather than having to change two versions in one place

Kichura committed:
Bump Quilt-JSON5 from 1.0.1 -> 1.0.2 in build.gradle.

EnnuiL commented on build.gradle:
```diff
@@ -11,8 +11,8 @@ plugins {
 	id 'org.quiltmc.gradle.licenser' version '1.1.+'
-	id 'org.quiltmc.loom' version '0.12.+' apply false
-	id 'com.github.johnrengelman.shadow' version '7.0.0'
+	id 'org.quiltmc.loom' version '0.13.+' apply false
```
0.13 isn't really supported due to it being essentially a beta to 1.0; Could you try that one?

Kichura replied:
Loom 0.13 seems to be working. (so far)

triphora replied:
What she's saying is to use 1.0 or 0.12, but not 0.13.

Kichura replied:
Undoed.

EnnuiL replied:
I'm still curious about the avoidance of Loom 1.0; Are there are major errors or?

Kichura replied:
I haven't tried it yet as i usually go from 0.12 -> 0.13 in ordering sense, although can test 1.0.

triphora commented on gradle.properties:
```diff
@@ -5,5 +5,5 @@ group = org.quiltmc
 url = https://github.com/quiltmc/quilt-loader
 
-asm_version = 9.3
+asm_version = 9.4
```
Have you tested that mixin supports this version of ASM?

Kichura replied:
I have tried this on a modded instance and it works fine with it.

Kichura replied:
> Have you tested that mixin supports this version of ASM?

It seems to work fine on client-side but not on server-side so i undid that as a result.

Kichura committed:
JUnit wants setupPaths() to be public.

AlexIIL commented on build.gradle:
```diff
@@ -1,7 +1,7 @@
-		classpath 'org.kohsuke:github-api:1.114'
-		classpath 'com.guardsquare:proguard-gradle:' + (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11) ? '7.2.0-beta2' : '7.1.0')
+		classpath 'org.kohsuke:github-api:1.313'
+		classpath 'com.guardsquare:proguard-gradle:' + (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11) ? '7.3.0-beta2' : '7.2.0')
```
Previously quilt-loader was buildable with java 8 through to 17, however some of the gradle plugins we use actually require java 17 now - so this whole statement can be removed and replaced with the latest version. (If we still allowed building with java 8 then you'd need to revert the `7.1.0` -> `7.2.0` version bump since AFAIK `7.2.0` requires java 11).

```suggestion
		classpath 'com.guardsquare:proguard-gradle:7.3.0-beta2')
```

AlexIIL replied:
Previously quilt-loader was buildable with java 8 through to 17, however some of the gradle plugins we use actually require java 17 now - so this whole statement can be removed and replaced with the latest version. (If we still allowed building with java 8 then you'd need to revert the `7.1.0` -> `7.2.0` version bump since AFAIK `7.2.0` requires java 11).

```suggestion
		classpath 'com.guardsquare:proguard-gradle:7.3.0-beta2')
```

Kichura committed:
ASM 9.4 is completely broken.

AlexIIL reviewed (changes requested):
Can you *also* update `src/main/resources/fabric-installer.launchwrapper.json`? I don't know if that is actually used anymore, but so long as it exists we should keep it up-to-date.

AlexIIL commented:
Out of interest, what was wrong with ASM 9.4? I assume we'll need to update to it *eventually*, so if updating now causes problems I'd quite like to know what they are.

Kichura committed:
Remove the Version_11 context.

As request of AlexIIL's change.

Co-authored-by: AlexIIL <AlexIIL@users.noreply.github.com>

Kichura commented:
> Out of interest, what was wrong with ASM 9.4? I assume we'll need to update to it _eventually_, so if updating now causes problems I'd quite like to know what they are.

That one caused the "org/objectweb/asm/commons/ClassRemapper" not found error to pop up out of thin air with no further explanation, guessing there is some code hidden somewhere i couldn't figure out on my own...?

AlexIIL commented on build.gradle:
```diff
@@ -1,7 +1,7 @@
-		classpath 'org.kohsuke:github-api:1.114'
-		classpath 'com.guardsquare:proguard-gradle:' + (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11) ? '7.2.0-beta2' : '7.1.0')
+		classpath 'org.kohsuke:github-api:1.313'
+		classpath 'com.guardsquare:proguard-gradle:7.3.0-beta2')
```
```suggestion
		classpath 'com.guardsquare:proguard-gradle:7.3.0-beta2'
```
Sorry, I didn't remove the extra bracket.

Kichura committed:
Remove the ).

As request of AlexIIL's change.

Co-authored-by: AlexIIL <AlexIIL@users.noreply.github.com>

Kichura committed:
Upstream gradle to 7.5.1.

Kichura committed:
Upstream fabric's launchwrapper.

EnnuiL commented:
Oh, one more note: With the Gradle update, you should also update the wrapper with `./gradlew wrapper`

Kichura committed:
Revert loom back to 0.12.+.

I thought 0.13 was a newer version of loom's previous version but it for some reason turned out to be 1.0 beta so...oops.

Kichura committed:
Update ubuntu to 22.04, Checkout to v3.

Kichura committed:
Upload new gradle-wrapper. (Attempt 1)

Kichura committed:
Upload new gradle-wrapper. (Attempt 2)

Kichura committed:
Upload new gradle-wrapper. (Attempt 3)

Trying to fix this via mobile alone is a hassle already lol.

Kichura commented:
> Oh, one more note: With the Gradle update, you should also update the wrapper with `./gradlew wrapper`

Think i managed to get it updated although done it via mobile with 3 attempts. 😅

Kichura committed:
I have no idea what to expect from this. (Loom 1.x)

Kichura reviewed (commented):
this works.

triphora commented:
> Think i managed to get it updated although done it via mobile with 3 attempts. :sweat_smile:

Nope, because the wrapper file is still identical. Do it on desktop.

Kichura committed:
The 4th time i tried to replace wrapper with new copy now.

Kichura commented:
> > Think i managed to get it updated although done it via mobile with 3 attempts. sweat_smile
> 
> Nope, because the wrapper file is still identical. Do it on desktop.

Darn - Thought i had it for real that time, will see if next commit makes any difference.

triphora commented:
It's worth noting that the gradlew and gradlew.bat files should have also been updated when you executed that command, but they were decidedly not.

Kichura commented:
> It's worth noting that the gradlew and gradlew.bat files should have also been updated when you executed that command, but they were decidedly not.

I did not update those because i thought they needed no changes - will go and do them now.

Kichura committed:
Upstream the gradlew scripts.

Kichura committed:
Let's try temurin as openjdk may be broken.

Kichura reviewed (commented):
Seems to fix the "xargs is not available" error as gradle 7.x has issues with oracle's java copies.

Kichura committed:
Move variables to gradle.properties.

While this is going to work, this however doesn't seem to work well inside the installer jsons at all - Although help for this might be useful.

Kichura committed:
Actually make syncing system work.

I asked Emmaffle for assistance - Although i think this can use some improvements.

Kichura commented:
> I would rather a system be implemented to automatically edit the `installer.json` file rather than having to change two versions in one place

Should be implemented although it can use improvements if any.

Kichura committed:
Don't duplicate sentences.

triphora commented on build.gradle:
```diff
@@ -111,6 +111,64 @@ processResources {
+	inputs.property "net.fabricmc:asm-util", project.asm
+	inputs.property "org.quiltmc:quilt-config", project.quilt_config
+
+	filesMatching("quilt_installer.json") {
```
```suggestion
	filesMatching("quilt_installer.json", "fabric-installer.launchwrapper.json") {
```

triphora commented on build.gradle:
```diff
@@ -111,6 +111,64 @@ processResources {
+	}
+}
+
+// Same as above but for the launchwrapper.
+processResources {
+	inputs.property "net.fabricmc:tiny-mappings-parser", project.tiny_mappings_parser
+	inputs.property "net.fabricmc:sponge-mixin", project.sponge_mixin
+	inputs.property "net.fabricmc:tiny-remapper", project.tiny_remapper
+	inputs.property "net.fabricmc:access-widener", project.access_widener
+	inputs.property "org.quiltmc:quilt-json5", project.quilt_json5
+	inputs.property "net.fabricmc:asm", project.asm
+	inputs.property "net.fabricmc:asm-analysis", project.asm
+	inputs.property "net.fabricmc:asm-commons", project.asm
+	inputs.property "net.fabricmc:asm-tree", project.asm
+	inputs.property "net.fabricmc:asm-util", project.asm
+	inputs.property "org.quiltmc:quilt-config", project.quilt_config
+
+	filesMatching("fabric-installer.launchwrapper.json") {
+		expand "net.fabricmc:tiny-mappings-parser": project.version
+		expand "net.fabricmc:sponge-mixin": project.version
+		expand "net.fabricmc:tiny-remapper": project.version
+		expand "net.fabricmc:access-widener": project.version
+		expand "org.quiltmc:quilt-json5": project.version
+		expand "net.fabricmc:asm": project.version
+		expand "net.fabricmc:asm-analysis": project.version
+		expand "net.fabricmc:asm-commons": project.version
+		expand "net.fabricmc:asm-tree": project.version
+		expand "net.fabricmc:asm-util": project.version
+		expand "org.quiltmc:quilt-config": project.version
+	}
+}
+
```
```suggestion
```
See other comment.

Kichura committed:
Add fabric-installer.launcherwrapper.json as additional file.

As request by wafflecoffee's changes.

Co-authored-by: wafflecoffee <emmaffle@modrinth.com>

Kichura committed:
Remove duplicated processResources.

As request by wafflecoffee's changes.

Co-authored-by: wafflecoffee <emmaffle@modrinth.com>

triphora commented on build.gradle:
```diff
@@ -111,6 +111,35 @@ processResources {
+		expand "net.fabricmc:asm-util": project.version
+		expand "org.quiltmc:quilt-config": project.version
+	}
+}
```
The way you've implemented this doesn't actually work. Like at all.

Kichura replied:
It's very rough indeed - might aswell go and find a better solution than this.

Kichura replied:
Fixed.

triphora replied:
No, it's not fixed, you don't know how processResources works. I will submit a PR to your PR.

Kichura committed:
Forgot to add "[" and "]".

Yes i know, This was copied & pasted; although this already shows how poor my programming skills are.

triphora committed:
Process resources fixes

Kichura committed:
Fix the process resources system entirely.

Thank you wafflecoffee for this amazing improvement!

EnnuiL reviewed (approved):
At this point? I feel like a migration to Gradle's version catalog system would be better; I'll approve it anyway

AlexIIL reviewed (commented):
This looks good - please don't change this anymore, I'd like to clone this branch as-is and test before merging.

Kichura commented:
> This looks good - please don't change this anymore, I'd like to clone this branch as-is and test before merging.

I will hold on to it as the installer change was a suggestion from oroarmor - if it is too much for a PR then feel free to undo it.

Kichura committed:
[ci skip] Upstream checkout to v3 on comment part.

Kichura committed:
[ci skip] Ensure commented container has temurin instead.

AlexIIL commented:
Okay, I've tested it now (finally :p) - If those are the last comments you want to add then I can merge now?

Kichura commented:
Yes this can be merged as i have checked everything else that needed upstreamings; as for ASM, can be resolved in another PR because i myself don't really know how to deal with that one.

AlexIIL commented:
Thanks for contributing!
